Controller/MediaAdminController.php

```diff
@@ -77,10 +77,12 @@ public function listAction(Request $request = null)
         $datagrid->setValue('context', null, $context);
 
         // retrieve the main category for the tree view
-        $category = $this->container->get('sonata.classification.manager.category')->getRootCategory($context);
+        $rootCategory = $this->container->get('sonata.classification.manager.category')->getRootCategory($context);
+        // This should be safe as the root category has to exist for a given context but I do not like fatal errors
```

Comment:
If I had to do it, I would do it on the CategoryManager level. But the actual code assumes the entity exists : 

```php
namespace Sonata\ClassificationBundle\Entity;

class CategoryManager
{
    /**
     * @param ContextInterface $context
     *
     * @return CategoryInterface
     */
    public function getRootCategory($context = null)
    {
        $context = $this->getContext($context);

        $this->loadCategories($context);

        return $this->categories[$context->getId()][0]; // Here we assume the entity always exists, with no verification at all
    }
```